maps/src/SingleDetectorMapBinner.cxx

```diff
@@ -0,0 +1,169 @@
+    std::deque<G3FramePtr> &out_queue)
+{
+	if (frame->Has(boloprops_name_))
+		boloprops_ = frame->Get<BolometerPropertiesMap>(
+		    boloprops_name_);
+
+	if (frame->type == G3Frame::EndProcessing) {
+		for (auto i : maps_) {
+			G3FramePtr out(new G3Frame(G3Frame::Map));
+			out->Put("Id", G3StringPtr(new G3String(i.first)));
+			out->Put("T", boost::dynamic_pointer_cast<G3FrameObject>
+			    (i.second.first));
+			out->Put("Wunpol", i.second.second);
+			out_queue.push_back(out);
+		}
+		maps_.clear(); // Don't sit on this memory anymore
+#ifdef OPENMP_FOUND
+		dets_.clear();
+#endif
+		out_queue.push_back(frame);
+		return;
+	}
+
+	if (frame->type != G3Frame::Scan) {
+		out_queue.push_back(frame);
+		return;
+	}
+
+	if (!boloprops_)
+		log_fatal("Need bolometer properties before detector data "
+		    "can be processed.");
+
+	G3VectorQuatConstPtr pointing =
+	    frame->Get<G3VectorQuat>(pointing_);
+	if (!pointing) {
+		log_error("Missing pointing %s", pointing_.c_str());
+		out_queue.push_back(frame);
+		return;
+	}
+
+	G3TimestreamMapConstPtr timestreams =
+	    frame->Get<G3TimestreamMap>(timestreams_);
+	if (!timestreams) {
+		log_error("Missing timestreams %s", timestreams_.c_str());
+		out_queue.push_back(frame);
+		return;
+	}
+	g3_assert(timestreams->NSamples() == pointing->size());
+
+	// Initialize units and maps on the first scan frame (when we have
+	// no in-progress maps).
+	if (maps_.empty()) {
+		template_->units = timestreams->GetUnits();
+		for (auto i : *timestreams) {
+			maps_[i.first].first = template_->Clone(false);
+			maps_[i.first].second = G3SkyMapWeightsPtr(
+			    new G3SkyMapWeights(template_, false));
+#ifdef OPENMP_FOUND
+			// Create a list of detectors to satisfy OpenMP's need
+			// for scalar iteration.
+			dets_.push_back(i.first);
+#endif
+		}
+	} else {
+		if (template_->units != timestreams->GetUnits())
+			log_fatal("Timestreams have units that do not match "
+			    "earlier timestreams in the pipeline.");
+	}
+
+
+#ifdef OPENMP_FOUND
+	#pragma omp parallel for
+	for (size_t i = 0; i < dets_.size(); i++) {
+		const std::string &det = dets_[i];
+		auto mw = maps_.at(det);
+		G3SkyMapPtr m = mw.first;
+		G3SkyMapPtr w = mw.second->TT;
+#else
+	for (auto i : maps_) {
+		const std::string &det = i.first;
+		G3SkyMapPtr m = i.second.first;
+		G3SkyMapPtr w = i.second.second->TT;
+#endif
+		G3TimestreamConstPtr ts = timestreams->at(det);
+	
+		// Get per-detector pointing timestream
+        	std::vector<double> alpha, delta;
+		auto bp = boloprops_->find(det);
+		if (bp == boloprops_->end())
+			log_fatal("Missing bolometer properties for %s",
+			    det.c_str());
+		get_detector_pointing(bp->second.x_offset, bp->second.y_offset,
+		    *pointing, m->coord_ref, alpha, delta);
+		auto pixels = m->AnglesToPixels(alpha, delta);
+
+		g3_assert(ts->size() == pixels.size());
+		for (size_t j = 0; j < ts->size(); j++) {
+			(*m)[pixels[j]] += (*ts)[j];
+			(*w)[pixels[j]] += 1;
```

Comment:
Do we want to allow things like PSD weights here?

Reply:
We could -- I'd like to keep feature enhancements for future versions of this code, however.